packages/design-toolkit/src/components/coordinate-field/index.tsx

```diff
@@ -0,0 +1,464 @@
+ *   errorMessage={errorMessage}
+ * />
+ */
+export function CoordinateField({ ref, ...props }: CoordinateFieldProps) {
+  [props, ref] = useContextProps(props, ref, CoordinateFieldContext);
+
+  const {
+    classNames,
+    description: descriptionProp,
+    label: labelProp,
+    format = 'dd',
+    size = 'medium',
+    enableCopy = true,
+    isDisabled = false,
+    isInvalid: isInvalidProp = false,
+    isRequired = false,
+    ...rest
+  } = props;
+
+  const isSmall = size === 'small';
+
+  const DOMProps = filterDOMProps(rest, { global: true });
+
+  // When size is small and label is hidden, use aria-label instead of aria-labelledby
+  // to ensure screen readers have an accessible name
+  const ariaLabelForSmallSize =
+    isSmall && labelProp ? labelProp : rest['aria-label'];
+
+  const {
+    state,
+    focus,
+    paste,
+    copy,
+    fieldProps,
+    labelProps,
+    descriptionProps,
+    errorProps,
+    validation,
+    effectiveErrorMessage,
+    isInvalid,
+  } = useCoordinateField(
+    props,
+    ariaLabelForSmallSize,
+    rest['aria-describedby'],
+    rest['aria-details'],
+  );
+
+  const componentState = {
+    segmentValues: state.segmentValues,
+    format,
+    currentValue: state.currentValue,
+    validationErrors: state.validationErrors,
+    isDisabled,
+    isInvalid,
+    isRequired,
+    size,
+  };
+
+  // Memoize all coordinate format conversions to avoid redundant calculations
+  const allCoordinateFormats = useMemo(
+    () => getAllCoordinateFormats(state.currentValue),
+    [state.currentValue],
+  );
+
+  // Calculate the maximum width needed for the control container
+  // This keeps the outlined container at a fixed width while segments animate
+  const maxControlWidth = useMemo(() => {
+    // Sum all editable segment max widths with their padding
+    const segmentWidth = state.editableSegmentConfigs.reduce((sum, config) => {
+      const maxLen = config.maxLength || 0;
+      const padding = config.pad ?? 0.5;
+      return sum + maxLen + padding;
+    }, 0);
+
+    // Calculate width of literal characters (colons, spaces, etc.)
+    const literalWidth = state.segmentConfigs
+      .filter((c) => c.type === 'literal')
+      .reduce((sum, c) => sum + (c.value?.length || 0), 0);
+
+    // Add gap spacing between segments (0.5rem = ~0.5ch per gap)
+    // Number of gaps = number of segments - 1
+    const gapCount = state.segmentConfigs.length - 1;
+    const gapWidth = gapCount * 0.5;
+
+    // Add container padding (px-s = 0.5rem on each side = ~1ch total)
+    const paddingWidth = 2;
+
+    // Add format button width (icon button is roughly 2.5-3ch)
+    const buttonWidth = enableCopy ? 3.5 : 0;
+
+    // Add gap between input and button (gap-m = 1rem = ~1ch)
+    const inputButtonGap = enableCopy ? 1.5 : 0;
+
+    return `${segmentWidth + literalWidth + gapWidth + paddingWidth + buttonWidth + inputButtonGap}ch`;
+  }, [state.editableSegmentConfigs, state.segmentConfigs, enableCopy]);
```

Comment:
This is a pretty beefy func with lots of "magic numbers" that could break if styles change. Consider pulling this out to a util